packages/stream-api/src/streamApi.ts

```diff
@@ -38,78 +43,125 @@ export class StreamApi {
 
   public startStream(
     data: vertexvis.protobuf.stream.IStartStreamPayload
-  ): Promise<void> {
+  ): Promise<void>;
+
+  public startStream(
+    requestId: UUID.UUID,
+    data: vertexvis.protobuf.stream.IStartStreamPayload
+  ): Promise<vertexvis.protobuf.stream.IStartStreamResult>;
+
+  public startStream(...args: unknown[]): unknown {
+    const request = this.parseArgs(args);
     return this.sendRequest({
+      ...request.requestId,
       startStream: {
-        ...data,
+        ...request.data,
       },
     });
   }
 
-  public beginInteraction(): Promise<void> {
+  public beginInteraction(): Promise<void>;
+
+  public beginInteraction(
+    requestId: UUID.UUID
+  ): Promise<vertexvis.protobuf.stream.IBeginInteractionResult>;
+
+  public beginInteraction(...args: unknown[]): unknown {
+    const request = this.parseArgs(args);
     return this.sendRequest({
+      ...request.requestId,
       beginInteraction: {},
     });
   }
 
   public replaceCamera({
     camera,
-  }: vertexvis.protobuf.stream.IUpdateCameraPayload): Promise<void> {
-    return this.sendRequest({
+  }: vertexvis.protobuf.stream.IUpdateCameraPayload): Promise<void>;
+
+  public replaceCamera(
+    requestId: UUID.UUID,
+    { camera }: vertexvis.protobuf.stream.IUpdateCameraPayload
+  ): Promise<vertexvis.protobuf.stream.IUpdateCameraResult>;
+
+  public replaceCamera(...args: unknown[]): unknown {
+    const request = this.parseArgs(args);
+    const requestPayload = {
+      ...request.requestId,
       updateCamera: {
-        camera,
+        camera: request.data as vertexvis.protobuf.stream.ICamera,
       },
-    });
+    };
+    return this.sendRequest(requestPayload);
   }
 
+  public hitItems({
+    point,
+  }: vertexvis.protobuf.stream.IHitItemsPayload): Promise<void>;
```

Comment:
unlikely this will be a call anyone wants to do, but providing makes the generic `sendRequest` simpler. will keep looking at alternatives.